witness/src/Dscp/Snowdrop/Expanders.hs

```diff
@@ -100,28 +112,42 @@ seqExpandersPublicationTx =
                 }
             } -> do
                 maybePub <- queryOne (PublicationsOf ptAuthor)
-
-                let change = if isJust maybePub then Upd else New
-
-                return $ mkDiffCS $ Map.fromList
-                  [ PublicationsOf  ptAuthor ==> change (LastPublication ptBlock)
-                  , PublicationHead ptBlock  ==> New    (PublicationNext ptPrevBlock)
-                  ]
+                queryOne (AccountId ptAuthor) >>= \case
+                    Just (account@ Account {..}) -> do
+                        let change = if isJust maybePub then Upd else New
+                        return $ mkDiffCS $ Map.fromList
+                          [ PublicationsOf  ptAuthor ==> change (LastPublication ptBlock)
+                          , PublicationHead ptBlock  ==> New    (PublicationNext ptPrevBlock)
+                          , AccountId ptAuthor ==>
+                                Upd account
+                                    -- You can say that `PubOf ==> LastPub`
+                                    -- mechanism will prevent double-pub
+                                    -- but lets not depend on that impl detail.
+                                    { aNonce   = aNonce + 1
+                                    , aBalance = aBalance - fromIntegral fee
+                                    }
+                          ]
+
+                    Nothing -> do
+                        throwLocalError PublicationAuthorDoesNotExist
 
 ----------------------------------------------------------------------------
 -- Money tx
 ----------------------------------------------------------------------------
 
-seqExpandersGTx ::
-       GTxWitnessed -> SeqExpanders Exceptions Ids Proofs Values ctx GTxWitnessed
-seqExpandersGTx (GMoneyTxWitnessed _) =
-    contramap (\(GMoneyTxWitnessed tx) -> tx) seqExpandersBalanceTx
-seqExpandersGTx (GPublicationTxWitnessed _) =
-    contramap (\(GPublicationTxWitnessed ptx) -> ptx) seqExpandersPublicationTx
+seqExpandersGTx
+    :: Word32
+    -> GTxWitnessed
+    -> SeqExpanders Exceptions Ids Proofs Values ctx GTxWitnessed
+seqExpandersGTx _ (GMoneyTxWitnessed _) =
+    flip contramap seqExpandersBalanceTx $ \(GMoneyTxWitnessed tx) -> tx
```

Comment:
I don't get it, why do we not use the `fee` parameter here?

Reply:
Fee is not needed in money transfer expander, since we're already doing changes in author account.
It is used from proof in validator.